### PR TITLE
Added support for `enumSuffix`

### PR DESCRIPTION
**Pull request by esfomeado**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

_Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request._

## Description

- Adds support for disabling the prefix on Enums similar to how `enumPrefix` works for prefixes.

Related #7839

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots/Sandbox (if appropriate/relevant):

Adding links to sandbox or providing screenshots can help us understand more about this PR and take action on it as appropriate

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Tests added on the PR

**Test Environment**:

- OS: Windows 11
- `@graphql-codegen/...`: Latest
- NodeJS: 19.9.0

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by surecloud-tstafford** [comment]
Any idea when this can be merged? Would really love this feature!

File: .changeset/swift-hornets-camp.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"@graphql-codegen/typescript-operations": patch
+"@graphql-codegen/typescript-resolvers": patch
+"@graphql-codegen/typescript": patch
+"@graphql-codegen/visitor-plugin-common": patch
+"website": patch
```
Comment (saihaj) [inline .changeset/swift-hornets-camp.md:6]:
```suggestion
"@graphql-codegen/typescript-operations": minor
"@graphql-codegen/typescript-resolvers": minor
"@graphql-codegen/typescript": minor
"@graphql-codegen/visitor-plugin-common": minor
```

File: packages/plugins/typescript/operations/tests/ts-documents.spec.ts [hunk 2]
```diff
@@ -2644,6 +2644,58 @@ describe('TypeScript Operations Plugin', () => {
       expect(o).toContain(`__typename?: 'Information_Entry', id: Information_EntryType,`);
     });
 
+    it('Should produce valid output with preResolveTypes=true and enums with suffixes set', async () => {
```
Comment (saihaj) [inline packages/plugins/typescript/operations/tests/ts-documents.spec.ts:2647]: this test we have suffix disabled 🤔 
```suggestion
    it('Should produce valid output with preResolveTypes=true and enums with no suffixes', async () => {
```

**Review by saihaj — APPROVED** [review]

**Review by beerose — APPROVED** [review]

**Comment by esfomeado** [comment]
Please let me know if you need anything else to get this merged.

Thanks

**Review by n1ru4l — APPROVED** [review]

**Pull request merged**